File: src/main/kotlin/wowsft/model/gameparams/ship/component/torpedo/Launcher.java
package wowsft.model.gameparams.ship.component.torpedo;

import wowsft.config.WoWSFT;
import wowsft.model.gameparams.TypeInfo;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;
import lombok.experimental.Accessors;

import java.util.List;

@Data
@WoWSFT
@Accessors(chain = true)
@JsonIgnoreProperties(ignoreUnknown = true)
public class Launcher
{
    private List<String> ammoList;
    private float barrelDiameter;
    private boolean canRotate;
    private List<List<Float>> deadZone;
    private List<Float> horizSector;
    private long id;
    private String index;
    private List<Float> mainSector;
    private String name;
    private int numAmmos;
    private int numBarrels;
    private List<Float> position;
    private List<Float> rotationSpeed;
    private List<Float> shootSector;
    private float shotDelay;
    private boolean smallGun;
    private float timeBetweenShots;
    private float timeToChangeAngle;
    private float timeToChangeSpeed;
    private List<Float> torpedoAngles;
    private TypeInfo typeinfo;
    private List<Float> vertSector;
}

File: src/main/kotlin/wowsft/model/gameparams/ship/component/torpedo/Launcher.kt
package wowsft.model.gameparams.ship.component.torpedo

import wowsft.config.WoWSFT
import wowsft.model.gameparams.TypeInfo
import com.fasterxml.jackson.annotation.JsonIgnoreProperties

@WoWSFT
@JsonIgnoreProperties(ignoreUnknown = true)
class Launcher {
    var ammoList = ArrayList<String>()
    var barrelDiameter = 0.0
    var canRotate = false
    var deadZone = ArrayList<MutableList<Double>>()
    var horizSector = ArrayList<Double>()
    var id: Long = 0
    var index: String? = null
    var mainSector = ArrayList<Double>()
    var name: String? = null
    var numAmmos = 0
    var numBarrels = 0
    var position = ArrayList<Double>()
    var rotationSpeed = ArrayList<Double>()
    var shootSector = ArrayList<Double>()
    var shotDelay = 0.0
    var smallGun = false
    var timeBetweenShots = 0.0
    var timeToChangeAngle = 0.0
    var timeToChangeSpeed = 0.0
    var torpedoAngles = ArrayList<Double>()
    var typeinfo : TypeInfo? = null
    var vertSector = ArrayList<Double>()
}
